### PR TITLE
sonar: fix possible NPE, same fix of svn/git provider

### PR DESCRIPTION
**Pull request by dfa1**

### CODE REVIEW

File: maven-scm-providers/maven-scm-provider-bazaar/src/main/java/org/apache/maven/scm/provider/bazaar/BazaarScmProvider.java [hunk 1]
```diff
@@ -81,7 +81,12 @@ public ScmProviderRepository makeProviderScmRepository( String scmSpecificUrl, c
     public ScmProviderRepository makeProviderScmRepository( File path )
         throws ScmRepositoryException, UnknownRepositoryStructure
     {
-        if ( path == null || !path.isDirectory() )
+        if ( path == null )
+        {
+            throw new NullPointerException( "Path argument is null" );
```
Comment (michael-o) [inline maven-scm-providers/maven-scm-provider-bazaar/src/main/java/org/apache/maven/scm/provider/bazaar/BazaarScmProvider.java]: Please say "path cannot be null".

File: maven-scm-providers/maven-scm-provider-hg/src/main/java/org/apache/maven/scm/provider/hg/HgScmProvider.java [hunk 2]
```diff
@@ -159,7 +159,12 @@ else if ( url.startsWith( "http" ) )
     public ScmProviderRepository makeProviderScmRepository( File path )
         throws ScmRepositoryException, UnknownRepositoryStructure
     {
-        if ( path == null || !path.isDirectory() )
+        if ( path == null )
+        {
+            throw new NullPointerException( "Path argument is null" );
```
Comment (michael-o) [inline maven-scm-providers/maven-scm-provider-hg/src/main/java/org/apache/maven/scm/provider/hg/HgScmProvider.java]: Please say "path cannot be null".

File: maven-scm-providers/maven-scm-providers-cvs/maven-scm-provider-cvs-commons/src/main/java/org/apache/maven/scm/provider/cvslib/AbstractCvsScmProvider.java [hunk 3]
```diff
@@ -199,7 +199,12 @@ public ScmProviderRepository makeProviderScmRepository( String scmSpecificUrl, c
     public ScmProviderRepository makeProviderScmRepository( File path )
         throws ScmRepositoryException, UnknownRepositoryStructure
     {
-        if ( path == null || !path.isDirectory() )
+        if ( path == null )
+        {
+            throw new NullPointerException( "Path argument is null" );
```
Comment (michael-o) [inline maven-scm-providers/maven-scm-providers-cvs/maven-scm-provider-cvs-commons/src/main/java/org/apache/maven/scm/provider/cvslib/AbstractCvsScmProvider.java]: Please say "path cannot be null".

File: maven-scm-providers/maven-scm-provider-bazaar/src/main/java/org/apache/maven/scm/provider/bazaar/BazaarScmProvider.java (hunk 1 above)
Comment (dfa1) [reply maven-scm-providers/maven-scm-provider-bazaar/src/main/java/org/apache/maven/scm/provider/bazaar/BazaarScmProvider.java]: I agree but please consider that other ScmProviders use "Path argument is null" (e.g. AbstractSvnScmProvider)

Comment (michael-o) [reply maven-scm-providers/maven-scm-provider-bazaar/src/main/java/org/apache/maven/scm/provider/bazaar/BazaarScmProvider.java]: Agreed, consistency is important. Leave as-is.

**Pull request closed without merging**